From a079edc4472c0b9f7abc9ae59867f032df3dbbc4 Mon Sep 17 00:00:00 2001
From: Terrence Pu <terrence.pu@amlogic.com>
Date: Mon, 8 Jun 2020 12:13:38 -0700
Subject: [PATCH 7/9] u-boot: Work around the regression block accessing
 Recovery

PD#SWPL-26780

Problem:
Bootloader enable the stack-protection-strong check and block RDK Recovery.
New features are added to support RDK Reovery.
It's too late to adjust the memory map which will increase bootloader stack size.
Thereofore we will just patch it.

Solution:
Retore to no-stack-protector only for entering RDK Recovery

Verify:
u212/HP40A

Change-Id: I63ad409283fec8a18f5b02d950bc2808d9e73bcb
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8f378bffb39..d4a9839a236 100644
--- a/Makefile
+++ b/Makefile
@@ -560,7 +560,7 @@ ifdef BUILD_TAG
 KBUILD_CFLAGS += -DBUILD_TAG='"$(BUILD_TAG)"'
 endif
 
-STACK_CFLAGS  := $(call cc-option,-fstack-protector-strong)
+STACK_CFLAGS  := $(call cc-option,-fno-stack-protector)
 KBUILD_CFLAGS += $(STACK_CFLAGS)
 
 KBUILD_CFLAGS	+= -g
-- 
2.31.1

